lib/rules/no-unused-vars.js

```diff
@@ -126,6 +131,30 @@ module.exports = {
         }
 
         /**
+         * Determines if a variable has a sibling rest property
+         * @param {Variable} variable - EScope variable object.
+         * @returns {boolean} True if the variable is exported, false if not.
+         * @private
+         */
+        function hasRestSpreadSibling(variable) {
+            if (config.ignoreRestSiblings) {
+                const restProperties = new Set(["ExperimentalRestProperty", "RestProperty"]);
+
+                return variable.defs
+                    .filter(def => def.name.type === "Identifier")
+                    .some(def => (
+                        def.node.id &&
+                        def.node.id.type === "ObjectPattern" &&
+                        def.node.id.properties.length &&
+                        restProperties.has(def.node.id.properties[def.node.id.properties.length - 1].type) &&  // last property is a rest property
+                        def.node.id.properties.some(prop => prop.type === "Property" && prop.key.name === variable.name)  // variable is sibling of the rest property
```

Comment:
I'm happy to change it, but I actually had purposefully wanted that example to be an error. My thinking was that you shouldn't rename a variable that's only used for omission. It would be slightly faster though to change it to check the parent type. Easy change, let me know :)